### PR TITLE
[Doc] Add descriptions for function runtime and context

### PR DESCRIPTION
**Pull request by Anonymitaet**
Document features introduced in https://github.com/apache/pulsar/pull/9445 and https://github.com/apache/pulsar/pull/9246.

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@freeznet could you please help review? Thanks.

**Review by freeznet — APPROVED** [review]
LGTM

File: site2/docs/functions-runtime.md [hunk 1]
```diff
@@ -278,7 +278,11 @@ The Kubernetes integration enables you to implement a class and customize how to
...
+
+Pulsar includes a built-in implementation. To use the basic implementation, set `runtimeCustomizerClassName` to `org.apache.pulsar.functions.runtime.kubernetes.BasicKubernetesManifestCustomizer`. The built-in implementation initialized with `runtimeCustomizerConfig` defined in the `functions-worker.yml` file enables you pass a JSON document as `customRuntimeOptions` with certain properties to augment to decide how the manifests are generated. If both `runtimeCustomizerConfig`  and `customRuntimeOptions` are provided, `BasicKubernetesManifestCustomizer` uses `customRuntimeOptions` to override the configuration if there are conflicts in these two configurations.
```
Comment (Huanli-Meng) [inline site2/docs/functions-runtime.md:283]: The built-in implementation initialized with `runtimeCustomizerConfig` defined in the `functions-worker.yml` file enables you pass a JSON document as `customRuntimeOptions` with certain properties to augment to decide how the manifests are generated.
For easy of read, it's better to break these long sentence into several short sentences.

**Review by Huanli-Meng — COMMENTED** [review]
Look Good to me and only leave a minor comment. PTAL

File: site2/docs/functions-runtime.md (hunk 1 above)
Comment (Jennifer88huang) [reply site2/docs/functions-runtime.md:283]: 1. I have the same concern on the long sentences.
2. "enables you pass..." --> "enables you to pass..."

**Review by sijie — APPROVED** [review]

File: site2/docs/functions-runtime.md (hunk 1 above)
Comment (Anonymitaet) [reply site2/docs/functions-runtime.md]: @Huanli-Meng @Jennifer88huang thanks for your comments. Updated, PTAL.

**Review by Huanli-Meng — APPROVED** [review]

**Pull request merged**